File: framework/src/main/java/com/thornbirds/frameworkext/component/Router.java
package com.thornbirds.frameworkext.component;

import androidx.annotation.NonNull;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Created by yangli on 2019/3/23.
 *
 * @mail yanglijd@gmail.com
 */
@Retention(RUNTIME)
@Target(TYPE)
public @interface Router {
    @NonNull String value();

    @NonNull String main() default "";
}

File: framework/src/main/java/com/thornbirds/frameworkext/component/route/Router.kt
package com.thornbirds.frameworkext.component.route

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @author YangLi yanglijd@gmail.com
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class Router(val value: String, val main: String = "")